src/Config/Factory.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PharIo\ComposerDistributor\Config;
+
+use PharIo\ComposerDistributor\File;
+use PharIo\ComposerDistributor\FileList;
+use PharIo\ComposerDistributor\Url;
+
+class Factory
+{
+	/**
+	 * Config json example:
+	 *
+	 * {
+	 *   "packageName": "phar-io/phive",
+	 *   "keyDirectory: "keys",
+	 *   "phars": [
+	 *     {
+	 *       "name": "phive",
+	 *       "file:" "https://github.com/phar-io/phive/releases/%version%/phive.phar",
+	 *       "signature:" "https://github.com/phar-io/phive/releases/%version%/phive.phar.asc"
+	 *     }
+	 *   ]
+	 * }
+	 *
+	 */
+	public static function createFromFile(string $configFile): Config
+	{
+		if (!is_file($configFile)) {
+			throw new \RuntimeException('Config file is missing');
+		}
+		$json = json_decode(file_get_contents($configFile), true);
+		if (json_last_error() !== JSON_ERROR_NONE) {
+			throw new \RuntimeException('Invalid json configuration');
+		}
```

Comment:
If we drop PHP 7.2 support - which will be EOL in 17 days - and instead require PHP 7.3+, we could use `json_decode(....., JSON_THROW_ON_ERROR);` rather than throwing a generic `RuntimeException`..

Reply:
Done